test/routes/home_route_test.dart

```diff
@@ -69,4 +72,75 @@ void main() {
       expect(ContentControl.instance.state.allSongs.songs, [songWith()]);
     });
   });
+
+  testWidgets('app shows exit confirmation toast', (WidgetTester tester) async {
+    await Prefs.confirmExitingWithBackButton.set(true);
+    final ToastObserver toastObserver = ToastObserver(tester);
+    final AppObserver appObserver = AppObserver(tester);
+    await tester.runAppTest(() async {
+      expect(
+        await appObserver.haveCloseRequest(
+          () => toastObserver.expectShowsToast(
+            BackButtonInterceptor.popRoute,
+            message: l10n.pressOnceAgainToExit,
+            reason: 'Expected the app to ask for confirmation before exiting',
+          ),
+        ),
+        isFalse,
+        reason: 'Expected the app not to close after showing the toast',
+      );
+      await tester.binding.delayed(const Duration(seconds: 5));
```

Comment:
Move the duration to constant in config.dart and test that exactly after that duration it allows us to quit the app

Reply:
Good idea, done.